Added sounds

MMCQ33 opened the pull request:
Added sounds and made them scaleable with volume. Added exclamationMark when ant finds resource

MMCQ33 committed:
Added sounds

Added sounds and made them scaleable with volume. Added exclamationMark when ant finds resource

ErikLouwerse commented on Hivemind/Assets/Scripts/Minds/Gathering.cs:
```diff
@@ -39,7 +39,7 @@ public enum State
     private int nextHarvest;
     private bool preparingReturn;
-
+    private GameObject excla;
```
Keep this is the function pls, no need for the whole class to know about it

ErikLouwerse commented on Hivemind/Assets/Scripts/TestScript.cs:
```diff
@@ -55,13 +56,16 @@ public void SpawnUnitForLastGroup()
-            var go = Instantiate(WorkerPrefab, SpawnPosition.position, Quaternion.identity,
-                GameObject.Find("Ants").transform);
-            var a = go.GetComponent<Ant>();
+            for (int x = 0; x < 20; x++)
```
pls no

ErikLouwerse commented on Hivemind/Assets/Scripts/TestScript.cs:
```diff
@@ -1,4 +1,5 @@
 ﻿using System;
+using System.Security.Cryptography.X509Certificates;
```
????

ErikLouwerse commented on Hivemind/Assets/Scripts/Minds/Gathering.cs:
```diff
@@ -165,11 +172,24 @@ public void Execute(Ant ant)
         }
     }
 
+    private void FoundResource()
```
Just put everything in the Discover function and start the coroutine where this function is currently called

MMCQ33 committed:
fixed comments and squashed a few bugs

Fixed pull request changes.
Fixed volume not updating correctly.
Fixed volume updating in each update.

ErikLouwerse commented on Hivemind/Assets/Scripts/Minds/Gathering.cs:
```diff
@@ -142,7 +142,13 @@ public void Execute(Ant ant)
                     if (ant.AtBase())
                     {
-                        if (IsScout && target != null) target.AddToKnownResourceList();
+                        if (IsScout && target != null)
```
make stacked if into a single one

ErikLouwerse commented on Hivemind/Assets/Scripts/Minds/Gathering.cs:
```diff
@@ -170,6 +176,14 @@ public double Likelihood(Ant ant)
+    {
+        ant.PlaySoundDiscovery();
+        var excla = (GameObject)GameObject.Instantiate(Resources.Load("ExclamationMark"), ant.transform, false);
+         excla.transform.localScale *= 3;
```
Spacing

MMCQ33 committed:
Fixed pause switching sometimes

fixed menu being switched and the game thinking it was in a different state than in the current state.

ErikLouwerse commented on Hivemind/Assets/Scripts/ResourceNode.cs:
```diff
@@ -119,4 +138,27 @@ public int GetResourcesFuture()
+            }
+
+            audioSrc.volume = audioVolume;
+            if (this.resourceType == ResourceType.Crystal)
```
useless this.

ErikLouwerse commented on Hivemind/Assets/Scripts/Ant.cs:
```diff
@@ -16,6 +16,8 @@ public class Ant : MonoBehaviour
     private Storage storage;
     internal Guid unitGroupID;
+    private AudioSource audioSrc;
+    private float audioVolume = 1f;
```
Why not just change the audioSrc.volume direly, this field seems redundant

ErikLouwerse replied:
REMOVE PLS

ErikLouwerse commented on Hivemind/Assets/Scripts/ResourceNode.cs:
```diff
@@ -31,6 +35,18 @@ private void Awake()
+            audioVolume = PlayerPrefs.GetFloat("Volume");
+        }
+        audioSrc.volume = (audioVolume);
+        if (this.resourceType == ResourceType.Crystal)
```
this.

ErikLouwerse commented on Hivemind/Assets/Scripts/ResourceNode.cs:
```diff
@@ -21,6 +21,10 @@ public class ResourceNode : MonoBehaviour
+    [SerializeField]
+    private AudioSource audioSrc;
+
+    private float audioVolume = 1;
```
Seems redundant, see other comment

MMCQ33 committed:
Fixed comments

Fixed pull request comments

MMCQ33 committed:
removed variable

rmoved unnecessary variable

ErikLouwerse commented on Hivemind/Assets/Scripts/Ant.cs:
```diff
@@ -107,9 +116,23 @@ internal void SetStorage(Storage storage)
+        {
+            audioSrc.volume = PlayerPrefs.GetFloat("Volume");
+        }
+        audioSrc.volume *=  0.05f;
```
this needs to be in the if, otherwise the audio will just decrease

MMCQ33 committed:
fixed comment

REEEEEEEEEEEEEE fixed comment
